manager/controllers/package_detail.go

```diff
@@ -0,0 +1,193 @@
+	var packageNames []models.PackageName
+	err := database.Db.Table("package_name").Take(&packageNames).Error
+	if err != nil {
+		return false
+	}
+	for _, pkg := range packageNames {
+		if pkg.Name == packageName {
+			return true
+		}
+	}
+	return false
+}
+
+type PackageDetailResponse struct {
+	Data PackageDetailItem `json:"data"`
+}
+
+type PackageDetailItem struct {
+	Attributes PackageDetailAttributes `json:"attributes"`
+	ID         string                  `json:"id"`
+	Type       string                  `json:"type"`
+}
+
+type PackageDetailAttributes struct {
+	Description string `json:"description"`
+	Summary     string `json:"summary"`
+	Name        string `json:"name"`
+	Version     string `json:"version"`
+	AdvisoryID  string `json:"advisory_id"`
+}
+
+func packageLatestHandler(c *gin.Context, packageName string) {
+	if !packageNameIsValid(packageName) {
+		c.JSON(http.StatusBadRequest, utils.ErrorResponse{Error: "invalid package name"})
+		return
+	}
+
+	query := singlePackageLatestQuery(packageName)
+
+	var packageInfo models.Package
+	err := query.Find(&packageInfo).Error
+	if err != nil {
+		LogAndRespNotFound(c, err, "package not found")
+		return
+	}
+
+	advisory, err := getAdvisoryByID(packageInfo.AdvisoryID)
+	if gorm.IsRecordNotFoundError(err) {
+		LogAndRespNotFound(c, err, "advisory not found")
+		return
+	}
+
+	summary, err := getPackageStrings(packageInfo.SummaryHash)
+	if gorm.IsRecordNotFoundError(err) {
+		LogAndRespNotFound(c, err, "summary not found")
+		return
+	}
+
+	descr, err := getPackageStrings(packageInfo.DescriptionHash)
+	if gorm.IsRecordNotFoundError(err) {
+		LogAndRespNotFound(c, err, "description not found")
+		return
+	}
+
+	nevra := packageName + "-" + packageInfo.EVRA
+
+	c.JSON(200, PackageDetailResponse{
+		Data: PackageDetailItem{
+			Attributes: PackageDetailAttributes{
+				Description: descr.Value,
+				Summary:     summary.Value,
+				Name:        packageName,
+				Version:     packageInfo.EVRA,
+				AdvisoryID:  advisory.Name,
+			},
+			ID:   nevra,
+			Type: "package",
+		},
+	})
+}
+
+func packageEvraHandler(c *gin.Context, nevra *utils.Nevra) {
+	if !packageNameIsValid(nevra.Name) {
+		c.JSON(http.StatusBadRequest, utils.ErrorResponse{Error: "invalid package name"})
+		return
+	}
+
+	packageName := nevra.Name
+	evra := nevra.EVRAString()
+
+	query := singlePackageEvraQuery(packageName, evra)
+
+	var packageInfo models.Package
+	err := query.Find(&packageInfo).Error
+	if err != nil {
+		LogAndRespNotFound(c, err, "package not found")
+		return
+	}
+
+	advisory, err := getAdvisoryByID(packageInfo.AdvisoryID)
+	if gorm.IsRecordNotFoundError(err) {
+		LogAndRespNotFound(c, err, "advisory not found")
+		return
+	}
+
+	summary, err := getPackageStrings(packageInfo.SummaryHash)
+	if gorm.IsRecordNotFoundError(err) {
+		LogAndRespNotFound(c, err, "summary not found")
+		return
+	}
+
+	descr, err := getPackageStrings(packageInfo.DescriptionHash)
+	if gorm.IsRecordNotFoundError(err) {
```

Comment:
Using correct sql query and joins you can load all info once into the `PackageDetailAttributes` object. It will be more efficient and shorter. Try it.
Also you can make basic part of query for both variants using the same method (joint package, package_name, advisory, strings). Then you should apply other criteria depending on whether it's latest (order by advisory pub.date) or exact version (where version is ...).